frameworks/keyed/ember/app/components/my-table.js

```diff
@@ -1,35 +1,101 @@
-import { inject as service } from '@ember/service';
 import Component from '@ember/component';
+import { action } from '@ember-decorators/object';
 
-export default Component.extend({
-  rows: service(),
-  actions: {
-    add: function() {
-      //console.log('add');
-      this.get('rows').add();
-    },
-    create: function() {
-      //console.log('create', this.rows);
-      this.get('rows').run();
-    },
-    update: function() {
-      //console.log('update', this.rows);
-      this.get('rows').update();
-    },
-    runLots: function() {
-      this.get('rows').runLots();
-    },
-    clear: function() {
-      this.get('rows').clear();
-    },
-    swapRows: function() {
-      this.get('rows').swapRows();
-    },
-    remove(identifier) {
-      this.get('rows').remove(identifier);
-    },
-    select(identifier) {
-      this.get('rows').select(identifier);
+import {
+  run, runLots, add, update, swapRows, deleteRow,
+} from 'ember-temp/utils/benchmark-helpers';
+
+var startTime;
+var lastMeasure;
+
+function startMeasure(name) {
+    startTime = performance.now();
+    lastMeasure = name;
+}
+function stopMeasure() {
+    var last = lastMeasure;
+
+    if (lastMeasure) {
+        window.setTimeout(function () {
+            lastMeasure = null;
+            var stop = performance.now();
+            console.log(last + " took " + (stop - startTime));
+        }, 0);
     }
+}
+
+export default class MyTable extends Component {
+  id = 1;
+  data = [];
+  selected = undefined;
+
+  didUpdate() {
+    stopMeasure();
+  }
+
+  @action async create() {
+    startMeasure('run');
+
+    const result = run(this.id);
+
+    this.setProperties({
+      data: result.data,
+      id: result.id,
+      selected: undefined
+    });
+  }
+
+  @action async add() {
+    startMeasure('add');
+
+    this.set('data', add(this.id, this.data).data);
+  }
+
+  @action async update() {
+    startMeasure('update');
+
+    this.set('data', update(this.data));
+  }
+
+  @action async runLots() {
+    startMeasure('runLots');
+
+    const result = runLots(this.id);
+
+    this.setProperties({
+      data: result.data,
+      id: result.id,
+      selected: undefined
+    });
+  }
+
+  @action async clear() {
+    startMeasure('clear');
+
+    this.setProperties({
+      data: [],
+      selected: undefined
+    });
+  }
+
+  @action async swapRows() {
+    startMeasure('swapRows');
+
+    this.set('data', swapRows(this.data));
+  }
+
+  @action async remove(id) {
```

Comment:
Probably don't use `async` without `await` or promises.

Reply:
the intent was to allow the UI to not be blocked due to the long running time of the benchmark helpers

Reply:
startMeasure and stopMeasure can be dropped completely, btw. they are not used by this benchmark.

Reply:
The real measurement happens with the test driver in webdriver-ts that uses chrome's timeline feature to measure the duration. Anything else is just an approximation (sometimes not too bad to validate the measurements).

Reply:
oh neat. ok, yeah, I'll remove all the measuring stuff.

Reply:
@krausest done